src/extension/gogoduck/src/main/java/au/org/emii/gogoduck/worker/GoGoDuckModule_cars.java

```diff
@@ -0,0 +1,12 @@
+package au.org.emii.gogoduck.worker;
+
+public class GoGoDuckModule_cars extends GoGoDuckModule {
```

Comment:
Yep. So that corresponds to the specialization you need for specific layers. So if you need specialization for `cars_2009`, you can just build:
```
public class GoGoDuckModule_cars_2009 extends GoGoDuckModule_cars
```

Then have specific behaviour for that. It's not very java-y, but I think it's not /that/ bad...